File: scripts/read_papers.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import pdf from 'pdf-parse/lib/pdf-parse.js';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const papersDir = path.join(__dirname, '../papers');

async function readPdfs() {
    const files = fs.readdirSync(papersDir).filter(f => f.endsWith('.pdf'));
    for (const file of files) {
        console.log(`\n\n--- Reading ${file} ---\n`);
        const dataBuffer = fs.readFileSync(path.join(papersDir, file));
        try {
            const data = await pdf(dataBuffer);
            console.log(data.text.substring(0, 2000)); // First 2000 chars to get an idea
        } catch (e) {
            console.error(`Error reading ${file}:`, e.message);
        }
    }
}

readPdfs();

File: scripts/read_papers.py
import os
import sys

papers_dir = os.path.join(os.path.dirname(__file__), '../papers')
output_file = os.path.join(os.path.dirname(__file__), 'papers_text.txt')

import fitz # PyMuPDF
with open(output_file, 'w', encoding='utf-8') as f:
    for file in os.listdir(papers_dir):
        if file.endswith('.pdf'):
            f.write(f"\n\n{'='*40}\n--- Reading {file} ---\n{'='*40}\n\n")
            try:
                doc = fitz.open(os.path.join(papers_dir, file))
                text = ""
                for page in doc[:3]: # First 3 pages of each
                    text += page.get_text()
                f.write(text)
            except Exception as e:
                f.write(f"Error: {e}\n")

print(f"Done reading. Output saved to {output_file}")
